packages/create-app/src/services/init.ts

```diff
@@ -73,15 +73,26 @@ async function init(options: InitOptions) {
           const packageJSON = (await findUpAndReadPackageJson(templateScaffoldDir)).content
           packageJSON.name = hyphenizedName
           packageJSON.author = (await username()) ?? ''
-          await updateCLIDependencies({packageJSON, local: options.local, directory: templateScaffoldDir})
+          packageJSON.private = true
+
+          switch (packageManager) {
+            case 'npm':
+            case 'yarn':
+              packageJSON.workspaces = ['web', 'web/frontend', 'extensions/*']
+              break
+            case 'pnpm':
+              await writeFile(
+                joinPath(templateScaffoldDir, 'pnpm-workspace.yaml'),
+                `packages:\n  - 'web'\n  - 'web/frontend'\n  - 'extensions/*'\n`,
+              )
+              // Ensure that the installation of dependencies doesn't fail when using
+              // pnpm due to missing peerDependencies.
+              await appendFile(joinPath(templateScaffoldDir, '.npmrc'), `auto-install-peers=true\n`)
```

Comment:
Regarding adding this to the templates, it would break any previous version of the CLI, that's why I added as code.
You can test this from main:
```bash
p create-app -n test-pnpm --path ~/tests --package-manager pnpm --template https://github.com/Shopify/shopify-app-template-node#workspaces-support
cd ~/tests/test-pnpm
pnpn install
pnpm dev
``` 

The process will crash unless you use `--skip-dependencies-installation`